scripts/okify_regtests

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+"""
+Regression test okifying script.
+
+Requires JFrog CLI (https://jfrog.com/getcli/) configured with credentials
+that have write access to the jwst-pipeline repository.
+"""
+
+import subprocess
+import os
+import tempfile
+from argparse import ArgumentParser
+from contextlib import contextmanager
+from glob import glob
+import shutil
+
+import asdf
+
+
+ARTIFACTORY_REPO = "jwst-pipeline-results"
+BUILD_NAME = "RT :: JWST"
+TERMINAL_WIDTH = shutil.get_terminal_size((80, 20)).columns
+
+
+def parse_args():
+    parser = ArgumentParser(description="Okify regression test results")
+    parser.add_argument("build_number", help="Jenkins build number for JWST builds", metavar="build-number")
+    parser.add_argument("--dry-run", action="store_true", help="pass the --dry-run flag to JFrog CLI")
+
+    return parser.parse_args()
+
+
+def artifactory_copy(specfile, dry_run=False):
+    jfrog_args = []
+
+    if dry_run:
+        jfrog_args.append("--dry-run")
+
+    args = list(["jfrog", "rt", "cp"]
+        + jfrog_args
+        + [f"--spec={specfile}"])
+    subprocess.run(args, check=True)
+
+
+def artifactory_get_okify_specfiles(build_number):
+    """Download specfiles from Artifactory associated with a build number and
+    return a list of their locations on the local file system
+
+    The search would be:
+
+    jfrog rt search jwst-pipeline-results/*/*_okify.json --props="build.number=540;build.name=RT :: JWST"
+    """
+    build_name = BUILD_NAME
+
+    props = f"build.number={build_number};build.name={build_name}"
+
+    # Retreive all the okify specfiles for failed tests.
+    args = list(["jfrog", "rt", "dl"]
+        + [f"{ARTIFACTORY_REPO}/*/*_okify.json"]
+        + [f'--props="{props}"']
+        + ["--flat"])
+    cmd = " ".join(args)
+    subprocess.run(cmd, check=True, shell=True, capture_output=True)
+
+    return sorted(glob("*_okify.json"))
+
+
+def artifactory_get_rtdata_asdf(build_number):
+    build_name = BUILD_NAME
+
+    props = f"build.number={build_number};build.name={build_name}"
+
+    # Retreive all the okify specfiles for failed tests.
+    args = list(["jfrog", "rt", "dl"]
+        + [f"{ARTIFACTORY_REPO}/*/*_rtdata.asdf"]
+        + [f'--props="{props}"']
+        + ["--flat"])
+    cmd = " ".join(args)
+    subprocess.run(cmd, check=True, shell=True, capture_output=True)
+
+    return sorted(glob("*_rtdata.asdf"))
+
+
+def artifactory_get_build_artifacts(build):
+    specfiles = artifactory_get_okify_specfiles(build)
+    asdffiles = artifactory_get_rtdata_asdf(build)
+
+    if len(specfiles) != len(asdffiles):
+        raise RuntimeError("Different number of _okify.json and _rtdata.asdf files")
+
+    for a,b in zip(specfiles, asdffiles):
+        if a.split(".")[0][:-6] != b.split(".")[0][:-7]:
```

Comment:
These index numbers are a little mysterious, may be worth defining constants:

```python
RTDATA_SUFFIX = "_rtdata.asdf"
SPECFILE_SUFFIX = "_okify.json"
```

And using those everywhere instead.